lib/pf/Switch/Fortinet/FortiGate.pm

```diff
@@ -247,7 +251,25 @@ sub returnAuthorizeVPN {
     my $kick = $self->handleRadiusDeny($args);
     return $kick if (defined($kick));
 
-    my $node = $args->{'node_info'};
+    my $role;
+    if ( isenabled($self->{_RoleMap}) && $self->supportsRoleBasedEnforcement()) {
```

Comment:
I have one potential concern.

Let's say you do CLI login authentication with a Catalyst switch and RoleMap is enabled, its going to return a Filter-Id for the CLI login from what I can tell.

Shouldn't we have a supportsVPNRoleBasedEnforcement method to handle this case specifically?

Or perhaps just let each switch module handle this in its returnRadiusVpn function?

Reply:
yes good point

Reply:
I have the VpnMap now so it´s an explicit switch config now.